### ISSUE TITLE
Write R script for automatically generating degree distribution charts from an adjacency matrix

### ISSUE THREAD

**Issue by kdahlquist** [kdahlquist/GRNmap#230]
Write R script for automatically generating degree distribution charts from an adjacency matrix.  @khorstmann and @maggie-oneil will work on a MATLAB implementation of this.

**Comment by kdahlquist** [comment]
@bklein7 and @Nwilli31 will work on this during the week of 9/19 (it goes along with the next chapters in the R tutorial #225.  We'll put the MATLAB version on hold for a while.  @Nwilli31 has done a bunch of these by hand, so she can communicate to @bklein7 what the script should do.  But essentially, it needs to be able to read the input network worksheet, compute the in-degree and out-degree for each gene and then make a bar chart.

**Comment by bklein7** [comment]
I communicated with @Nwilli31 to determine how this script should function and wrote a [preliminary version](https://github.com/kdahlquist/DahlquistLab/blob/master/R_scripts/Degree-Distribution-Generator.R). Upon testing, I was able to get this script to successfully output degree distribution charts according to @Nwilli31's parameters. However, I did not have enough time to complete this script. Changes I intend to implement next week are commented out beginning at line 5.

If you would like to test this version of the script, I recommend starting by saving an output from the random network generator. To do such a test, follow this procedure:

1. Invoke the  [random network generator script](https://github.com/kdahlquist/DahlquistLab/blob/master/R_scripts/Random-Matrix.R).
2. Type the following into R: `random.network=random.network(TARGETS,REGULATORS,EDGES)`
(replace the capitalized words with their corresponding integer values).
3. Invoke the [degree distribution generator script](https://github.com/kdahlquist/DahlquistLab/blob/master/R_scripts/Degree-Distribution-Generator.R).
4. Type the following into R exactly as written: `degree.distribution(random.network)`

**Comment by kdahlquist** [comment]
If there is time this week, @bklein7 and @Nwilli31 will make degree distribution charts for the "real" networks and store in a PowerPoint on Dahlquist Lab repository.  Also if there is time, generate random networks corresponding to the "real" ones (don't do dZAP1 until it's been vetted); and make the corresponding charts.

**Comment by kdahlquist** [comment]
Attempted to run script using instructions from  @bklein7's comment from Sep 21, 2016 and got an error.  @bklein7 and @kdahlquist should sit down together briefly and walk through running this so that this issue can be closed.

Also, need to verify if the degree distribution charts for the six db-derived and the random networks have been made.

**Comment by kdahlquist** [comment]
It would be interesting to see if we could use this paper to generate the random networks and compare to the db-derived and the random ones that @bklein7 generated.

Iorio, F., Bernardo-Faura, M., Gobbi, A., Cokelaer, T., Jurman, G., & Saez-Rodriguez, J. (2016). Efficient randomization of biological networks while preserving functional characterization of individual nodes. BMC bioinformatics, 17(1), 542. https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1402-1

I asked @Nwilli31 to include it in her annotated bibliography for her thesis.

**Comment by bklein7** [comment]
When could @kdahlquist meet to review the degree distribution chart script with me? Wednesday between 11:20am-1:00pm and Friday between 11:20am-2:30pm would work well for me if we are to meet prior to spring break.

**Comment by kdahlquist** [comment]
@bklein7, Wednesday is a no-go for me.  However, Friday between 11:20 and 1:30 would work.  I'm waiting on @anuvarsh to confirm a Friday meeting during that time frame as well, but we could tentatively say that we would either be meeting at 11:20 or at ~12:30.

I'll confirm when I know more.

**Comment by kdahlquist** [comment]
My schedule has been settled; let's do 11:20 on Friday for a meeting between @bklein7 and @kdahlquist.

**Comment by kdahlquist** [comment]
Closing this out because @bklein7 and @kdahlquist met and wrote instructions for how to run.  @kdahlquist will clean up instructions a little, but it is good to go.

**Comment by kdahlquist** [comment]
Instructions cleaned up and closing now.
